src/main/drivers/gpio.h

```diff
@@ -110,9 +110,9 @@ typedef struct
 static inline void digitalHi(GPIO_TypeDef *p, uint16_t i) { p->BSRR = i; }
 static inline void digitalLo(GPIO_TypeDef *p, uint16_t i)     { p->BRR = i; }
 static inline void digitalToggle(GPIO_TypeDef *p, uint16_t i) { p->ODR ^= i; }
-static inline uint16_t digitalIn(GPIO_TypeDef *p, uint16_t i) {return p->IDR & i; }
+static inline uint16_t digitalIn(const GPIO_TypeDef *p, uint16_t i) {return p->IDR & i; }
```

Comment:
yeah, this change doesn't fit with the rest of the digital* functions.  I agree that the gpio_config changes should be marked as const though.